File: leet-code/Algorithms/028-implement-strStr/sol_a.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        
        if len(needle) == 0:
            return 0
        
        for i in range(0, len(haystack)):
            # No point in trying if it goes past the end of the haystack
            if i+len(needle) > len(haystack):
                break
            
            good = False
            subHay = haystack[i:i+len(needle)]
            if subHay == needle:
                return i
                
                
        return -1

File: leet-code/Algorithms/028-implement-strStr/sol_a.rs
impl Solution {
    pub fn str_str(haystack: String, needle: String) -> i32 {
        if needle.len() == 0 {
            return 0;
        }
        
        for i in 0..haystack.len() {
            if i+needle.len() > haystack.len(){
                break;
            }
            if haystack[i..i+needle.len()] == needle {
                return i as i32;
            }
        }
        
        return -1;
    }
}
